apiserver/facades/client/modelmanager/modelmanager_test.go

```diff
@@ -303,6 +308,77 @@ func (s *modelManagerSuite) setAPIUser(c *gc.C, user names.UserTag) {
 	s.api = mm
 }
 
+// expectCreateModel expects all the calls to the services made during model
+// creation. It generates the calls based off the modelCreateArgs.
+func (s *modelManagerSuite) expectCreateModel(
+	c *gc.C,
+	ctrl *gomock.Controller,
+	modelCreateArgs params.ModelCreateArgs,
+	expectedCloudCredential credential.Key,
+	expectedCloudName string,
+	expectedCloudRegion string,
+) coremodel.UUID {
+	modelUUID := modeltesting.GenModelUUID(c)
+	userTag, err := names.ParseUserTag(modelCreateArgs.OwnerTag)
+	c.Assert(err, gc.IsNil)
+	ownerName := user.NameFromTag(userTag)
+	ownerUUID := usertesting.GenUserUUID(c)
+
+	// Get the default cloud name and credential.
+	s.modelService.EXPECT().DefaultModelCloudNameAndCredential(
+		gomock.Any()).Return("dummy", credential.Key{}, nil)
+	// Get the uuid of the model owner.
+	s.accessService.EXPECT().GetUserByName(
+		gomock.Any(), ownerName,
+	).Return(user.User{UUID: ownerUUID}, nil)
+
+	// Create model in controller database.
+	s.modelService.EXPECT().CreateModel(gomock.Any(), model.ModelCreationArgs{
+		Name:        modelCreateArgs.Name,
+		Owner:       ownerUUID,
+		Cloud:       expectedCloudName,
+		CloudRegion: expectedCloudRegion,
+		Credential:  expectedCloudCredential,
+	}).Return(
+		modelUUID,
+		func(context.Context) error { return nil },
+		nil,
+	)
+
+	// Create and setup model in model database.
+	s.expectCreateModelOnModelDB(ctrl, modelCreateArgs.Config)
+
+	// Called as part of getModelInfo which returns information to the user
+	// about the newly created model.
+	s.accessService.EXPECT().GetModelUsers(gomock.Any(), gomock.Any(), gomock.Any())
+
+	return modelUUID
+}
+
+// expectCreateModelOnModelDB expects all the service calls to the new model's
+// own database.
+func (s *modelManagerSuite) expectCreateModelOnModelDB(
+	ctrl *gomock.Controller,
+	modelConfig map[string]any,
+) {
+	// Expect call to get the model service factory.
+	modelServiceFactory := mocks.NewMockModelServiceFactory(ctrl)
+	s.serviceFactoryGetter.EXPECT().ServiceFactoryForModel(gomock.Any()).Return(modelServiceFactory)
+
+	// Expect calls to get various model services.
+	modelInfoService := mocks.NewMockModelInfoService(ctrl)
+	networkService := mocks.NewMockNetworkService(ctrl)
+	modelConfigService := mocks.NewMockModelConfigService(ctrl)
+	modelServiceFactory.EXPECT().ModelInfo().Return(modelInfoService)
+	modelServiceFactory.EXPECT().Network().Return(networkService)
+	modelServiceFactory.EXPECT().Config().Return(modelConfigService)
+
+	// Expect calls to functions of the model services.
+	modelInfoService.EXPECT().CreateModel(gomock.Any(), s.controllerUUID)
+	modelConfigService.EXPECT().SetModelConfig(gomock.Any(), modelConfig)
+	networkService.EXPECT().ReloadSpaces(gomock.Any())
```

Comment:
You might want to put `.AnyTimes()` after these expectations, otherwise you're asserting that they are called exactly once.

Reply:
For these I think its fine to expect exactly once. In create model they should all only be called once, if a retry is added or something then this can be updated in the tests